### ISSUE TITLE
Throughput changes result in Copy Volume instead of 

### ISSUE THREAD

**Issue by blyles** [cloudfoundry/bosh-aws-cpi-release#137]
We've been testing the resize capabilities of the cpi and came across an interesting problem.

Goal: 

We had a 4TB disk that we wanted to bump to 1000MB throughput on a GP3.

Outcome of changing 

```
    cloud_properties:
-     throughput: 500
+     throughput: 1000
```

Was that a **volume copy kicked off**, I think I was hoping that it'd follow that magic path of the resize capabilities and provide the throughput configuration but it didn't.

With the resize feature the idea that i was grasping onto is that it modifies the volume and continues saving us from the x hours of waiting on large disks to move around.  Looking at this;

https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EC2/Types/ModifyVolumeRequest.html

It suggests that providing throughput and iops independent of a resize could result in a management ease of iops and performance changes.  Would this be logical in a place like the CPI to do this?

As we may want to tune up our configs from a later date and spending X days just to up the throughput seems wrong.

**Comment by ramonskie** [comment]
did you enable the native resize option with directory property `enable_cpi_resize_disk`

**Comment by ramonskie** [comment]
any updates on this?

**Comment by blyles** [comment]
I have a few iterations of this locally I've just not had the time to spend on it, but i managed to get something working in the area of "if the size is the same and the settings are different it uses the aws api to natively change it", kind of like the way the resize works right now.  But haven't been able to get it working all the way.

**Comment by rkoster** [comment]
The best way to implement this would be to add a _update_disk_ cpi method similar to [resize_disk](https://bosh.io/docs/cpi-api-v2-method/resize-disk/) which should be called when the cloud_properties for a disk change. The cpi should validate the incoming cloud properties and determine if an in-place update is supported, if not it should return a not supported error to the director at which point the director should fall back to the original recreate disk behavior.

All of this would be a lot of work and should be discussed in more detail in a new issue to be created against the bosh director. I'm going to close this issue for now, due to inactivity.
